registry/service_discovery.go

```diff
@@ -34,6 +38,12 @@ type ServiceDiscovery interface {
 
 	// ----------------- lifecycle -------------------
 
+	/**
+	 * Initializes the  ServiceDiscovery
+	 *
+	 */
+	Initialize(registryURL common.URL) error
```

Comment:
u add this func for `EventPublishingServiceDiscovery`. and then u add it to ServiceDiscovery interface. Pls think again.

what's more, the func name is so long which should be `Init`.